Source/DataSources/RelativeToTerrainHeightProperty.js

```diff
@@ -0,0 +1,201 @@
+     * @param {Property} [positions] A Property specifying the {@link PolygonHierarchy} or an array of {@link Cartesian3} positions.
+     * @param {Property} [heightRelativeToTerrain] A Property specifying the numeric height value relative to terrain
+     *
+     * @example
+     * var hierarchy = new Cesium.ConstantProperty(polygonPositions);
+     * var redPolygon = viewer.entities.add({
+     *     polygon : {
+     *         hierarchy : hierarchy,
+     *         material : Cesium.Color.RED,
+     *         height : new Cesium.RelativeToTerrainHeightProperty(hierarchy, 11.0)
+     *     }
+     * });
+     */
+    function RelativeToTerrainHeightProperty(terrainProvider, positions, heightRelativeToTerrain) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('terrainProvider', terrainProvider);
+        //>>includeEnd('debug');
+
+        this._positions = undefined;
+        this._subscription = undefined;
+        this._heightRelativeToTerrain = undefined;
+        this._heightRelativeToTerrainSubscription = undefined;
+        this._definitionChanged = new Event();
+
+        this._terrainProvider = terrainProvider;
+        this._cartographicPosition = new Cartographic();
+        this._terrainHeight = 0;
+
+        this.positions = positions;
+        this.heightRelativeToTerrain = heightRelativeToTerrain;
+    }
+
+    defineProperties(RelativeToTerrainHeightProperty.prototype, {
+        /**
+         * Gets a value indicating if this property is constant.
+         * @memberof RelativeToTerrainHeightProperty.prototype
+         *
+         * @type {Boolean}
+         * @readonly
+         */
+        isConstant : {
+            get : function() {
+                return Property.isConstant(this._positions);
+            }
+        },
+        /**
+         * Gets the event that is raised whenever the definition of this property changes.
+         * @memberof RelativeToTerrainHeightProperty.prototype
+         *
+         * @type {Event}
+         * @readonly
+         */
+        definitionChanged : {
+            get : function() {
+                return this._definitionChanged;
+            }
+        },
+        /**
+         * Gets or sets the positions property used to compute the value.
+         * @memberof RelativeToTerrainHeightProperty.prototype
+         *
+         * @type {Property}
+         */
+        positions : {
+            get : function() {
+                return this._positions;
+            },
+            set : function(value) {
+                var oldValue = this._positions;
+                if (oldValue !== value) {
+                    if (defined(oldValue)) {
+                        this._subscription();
+                    }
+
+                    this._positions = value;
+
+                    if (defined(value)) {
+                        if (!value.isConstant) {
+                            throw new RuntimeError('positions must be a constant property');
```

Comment:
Shouldn't this be a developer error?

Reply:
We actually could support time-dynamic values if we just made it "best effort", i.e. it would only call `sampleTerrain` if a current request wasn't already in flight.